osquery/config/plugins/tests/tls_config_tests.cpp

```diff
@@ -0,0 +1,175 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <gtest/gtest.h>
+
+#include <vector>
+
+#include <boost/property_tree/ptree.hpp>
+
+#include <osquery/config.h>
+#include <osquery/database.h>
+#include <osquery/system.h>
+#include <osquery/tables.h>
+
+#include "osquery/core/conversions.h"
+#include "osquery/core/json.h"
+#include "osquery/remote/requests.h"
+#include "osquery/remote/serializers/json.h"
+#include "osquery/remote/transports/tls.h"
+#include "osquery/remote/utility.h"
+#include "osquery/tests/test_util.h"
+
+#include "osquery/config/plugins/tls.h"
+
+namespace pt = boost::property_tree;
+
+namespace osquery {
+
+DECLARE_string(tls_hostname);
+DECLARE_bool(config_tls_enroll_on_startup);
+
+namespace tables {
+QueryData genOsqueryInfo(QueryContext& context);
+}
+
+class TLSConfigTests : public testing::Test {};
+
+TEST_F(TLSConfigTests, test_setup) {
```

Comment:
The main motivation of adding the tls_config_tests was to test the behavior of the config_tls_enroll_on_startup flag, which was newly added as part of this PR